book/src/contributing/using-dev-container.md

```diff
@@ -0,0 +1,55 @@
+# Development in Dev Containers
+
+```admonish note
+This document is a heavy work in progress and may not contain sufficient information.
+```
+
+[Dev Containers](https://containers.dev/) are a way to package a number of
+"developer tools" (compilers, bundlers, package managers, loaders, etc.) into a
+single object. This is helpful when many people want to contribute to a project:
+each person only has to install the (single) Dev Container on their own machine
+to start working. By definition, the Dev Container has a consistent set of tools
+that are known to work together. This avoids a fuss with finding the proper
+versions of each of the build tools.
+
+To use a Dev Container on your local computer with VS Code, you must install the
+[VS Code Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+extension and its system requirements.
+
+When you first start the Dev Container, the build process may take a long time
+(as much as 20-40 minutes) as the container collects all the packages of the
+full PRQL toolchain. Once it's running, though, startup is fairly fast.
+
+## How you use it
+
+1. Clone the git repo onto your hard drive as usual. Then you start the Dev
+   Container (say, using VS Code) that bundles all the developer tools.
+2. Edit files locally (say, using VS Code) to work on PRQL. The tools in the Dev
+   Container watch for changed files and rebuild the project as needed.
```

Comment:
Is there actually watching going on? I think the volumes are mapped in, but there's no "watch" task, as it would be commonly known as.

(btw this is an example where just linking to docs is easier — can we actually explain this better than VSCode itself can?)

Reply:
I believe that when the Rust source code changes, rust-analizer automatically builds it.

It is not devcontainer specific, so I agree that it just doesn't need to be explained here.